### ISSUE TITLE
Documents not summarizing

### ISSUE THREAD

**Issue by andre4i** [microsoft/FluidFramework#7518]
We have received [another incident](https://portal.microsofticm.com/imp/v3/incidents/details/262156364/home) of a document accumulating a large number (40k) of OPs without summarization. The ICM incident contains a Kusto query for the document.

We are seeing a lot of:

```
fluid:telemetry:DeltaManager:WrapError	Nack: Submit a summary before inserting additional operations		Nack: Submit a summary before inserting additional operations
fluid:telemetry:Container:ContainerWarning	Service busy/throttled: Nack: Submit a summary before inserting additional operations		Service busy/throttled: Nack: Submit a summary before inserting additional operations
fluid:telemetry:DeltaManager:WrapError	Nack: Submit a summary before inserting additional operations		Nack: Submit a summary before inserting additional operations
fluid:telemetry:Container:ContainerWarning	Service busy/throttled: Nack: Submit a summary before inserting additional operations		Service busy/throttled: Nack: Submit a summary before inserting additional operations
fluid:telemetry:Container:ContainerWarning	Service busy/throttled: Nack: Submit a summary before inserting additional operations		Service busy/throttled: Nack: Submit a summary before inserting additional operations
fluid:telemetry:DeltaManager:WrapError	Nack: Submit a summary before inserting additional operations		Nack: Submit a summary before inserting additional operations
fluid:telemetry:DeltaManager:WrapError	Nack: Submit a summary before inserting additional operations		Nack: Submit a summary before inserting additional operations
```

And the summarizer logs show the following pattern:

```
fluid:telemetry:SummaryManager:CreatingSummarizer		
fluid:telemetry:Summarizer:StoppingSummarizer		parentNotConnected
(...)
fluid:telemetry:SummaryStatus:Behind		
fluid:telemetry:SummaryManager:CreatingSummarizer		
fluid:telemetry:Summarizer:StoppingSummarizer		parentNotConnected
(...)
fluid:telemetry:SummaryManager:CreatingSummarizer		
fluid:telemetry:Summarizer:StoppingSummarizer		parentNotConnected
fluid:telemetry:SummaryStatus:Behind		
fluid:telemetry:Summarizer:NotStarted		DifferentComputedSummarizer
fluid:telemetry:SummaryManager:RunningSummarizer_end		
(...)
fluid:telemetry:SummaryManager:CreatingSummarizer		
fluid:telemetry:SummaryStatus:Behind		
fluid:telemetry:Summarizer:StoppingSummarizer		parentShouldNotSummarize
fluid:telemetry:SummaryStatus:Behind		
fluid:telemetry:SummaryManager:CreatingSummarizer		
fluid:telemetry:Summarizer:StoppingSummarizer		parentNotConnected
```

Incident [257910811](https://portal.microsofticm.com/imp/v3/incidents/details/257910811/home) may be related, however that particular instance recovered by itself.

cc @vladsud @pleath

**Comment by vladsud** [comment]
First, some related issue I've been looking yesterday, with deeper investigation into 2 recent docs that hit similar problems: https://github.com/microsoft/FluidFramework/issues/7589

When it comes to docId == "M%2F1pui%2FeRmFe%2FD8nLpqwO1YhTAYYENnGQUOL4g8Kx%2B4%3D", here are some observations:

First events started on 2021-08-31, for loaderId == "14e785a3-5e6e-45fd-838c-7f126c00d931", runtimeVersion = 0.46.0
What I see is that whenever interactive client disconnects due to 10K ops nack, summarizer client does not close and keeps loading just fine, but then eventually when it fully catches up ("connects"), it closes itself with fluid:telemetry:Summarizer:NotStarted	DifferentComputedSummarizer
I.e. fluid:telemetry:Summarizer:StoppingSummarizer / parentNotConnected event that fires before tells us that this summarizer should go away, but it proceeds forward despite of it to evaluate if last summary needs to be performed (which gets us out of this state), but it never happens as by the time summarizing client is fully up to date, it bails out due to DifferentComputedSummarizer check.

This query looks at overall picture - both successes & failures:

> union Office_Fluid_FluidRuntime_*
| where Data_docId == "M%2F1pui%2FeRmFe%2FD8nLpqwO1YhTAYYENnGQUOL4g8Kx%2B4%3D"
| where Data_clientType == "noninteractive/summarizer"
| where Data_reason != "" or Data_error != "" or Data_eventName contains "Summarize_end" or Data_eventName contains "Summarize_cancel"
| where Data_eventName notcontains "DeltaManager"
| where Data_eventName notcontains "Container"
| project Event_Sequence, Event_Time, Data_eventName, Data_reason, Data_error, Data_clientType, Data_containerId

What I see from here is
1. For most part, when summaries are not going, parentNotConnected is followed by DifferentComputedSummarizer, and the later is the core reason we should look into why summaries are not being created.
   - Note that DifferentComputedSummarizer keyword does not exist in main any more. Looking through PRs, I've removed this code in https://github.com/microsoft/FluidFramework/pull/7230/files, which points to # 4 point in description of this PR on why this check was wrong and caused last Summary to not run properly - exactly the case we observer here!
2. There are cases where parentNotConnected is not followed by DifferentComputedSummarizer, but these cases are rarer, and it feels like it should be lower priority case to look into. For example, there is a block of such cases at the end (after 2021-09-07T17:29:35.75Z)
   - That said, if you look at nacks, you will see that there are a lot of nacks on 09/14. And looking at details (loaderId == "8adea59a-a1ec-4d1d-8855-8ff7ba60148c") you see that summarizer client was not attempted to be created for very long time after last summarizer quite. This looks to be https://github.com/microsoft/FluidFramework/pull/7321 problem. I checked - it is not in 0.46!

So to sum up, we need latest fixes. I have not checked, but I believe all of them are on 0.47 (need to double check), and all these sessions are for runtime = 0.46, so we are missing all of the proactive work I've done in this space.

**Comment by pleath** [comment]
This seems to be related to the old 10K ops issue, and possible fixed?
